sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -1264,7 +1264,14 @@ protected virtual LastEnqueuedEventProperties ReadLastEnqueuedEventProperties(st
                     // Canceling the main source here won't cause a problem and will help expedite stopping
                     // the processor later.
 
-                    _runningProcessorCancellationSource?.Cancel();
+                    try
+                    {
+                        _runningProcessorCancellationSource?.Cancel();
```

Comment:
Ah I missed the start processing cancellation in SB, though thinking about it I'm not sure how this would ever not be null.

Reply:
It never should if things are working correctly.  The only time that would happen is if Stop were broken and left an inconsistent state.

Reply:
Do we not set to null in a finally block in stop?

Reply:
We do not.   _([ref](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs#L1365))_